### ISSUE TITLE
Merging XLNet NER project into main repo?

### ISSUE THREAD

**Issue by stevezheng23** [zihangdai/xlnet#68]
Here is the XLNet extension project which includes a XLNet-NER implementation, [https://github.com/stevezheng23/xlnet_extension_tf](https://github.com/stevezheng23/xlnet_extension_tf).

This XLNet extension project is currently importing zihangdai/xlnet repo as its submodule, maybe we can consider merging it into the main repo?

**Comment by kimiyoung** [comment]
Great. It looks like the results on NER are a bit behind the current SoTA, which is over 93. It would be great to see whether the hparams or implementation could be improved.

**Comment by stevezheng23** [comment]
Yes, the result is from the initial run, I haven't tuned the hparams yet.

**Comment by stefan-it** [comment]
@stevezheng23 I've one question regarding to the NER implementation: have you also experiment with using different layers? E.g. the BERT paper (table 7) uses a feature-based approach with a concatenation of the last four layers. Could you give some details what layers you're using in your repo? Thanks :)

**Comment by stevezheng23** [comment]
@stefan-it In the initial experiments, I just finetuned the XLNet model by adding a dense + softmax layer on top of the last layer. For feature-based approach, I have't done corresponding experiments yet.

**Comment by mcggood** [comment]
@stefan-it One off-topic question please, in "concatenation of the last four layers", does "last four layers" means 9,10,11,12 layer?

**Comment by stefan-it** [comment]
@mcggood Yes :) Btw: here are the results for the feature-based approach from the BERT paper:

![image](https://user-images.githubusercontent.com/20651387/60434537-a21ca080-9c07-11e9-8223-f48b0eca73c8.png)
